usr/share/rear/lib/global-functions.sh

```diff
@@ -46,6 +56,23 @@ function is_positive_integer () {
     return 1
 }
 
+# Test if the (first) argument is a nonnegative integer (i.e. test for '>= 0')
+# If yes output the argument value and return 0
+# otherwise output '0' and return 1
+# (in particular "is_nonnegative_integer -00" outputs '0' and returns 0):
+function is_nonnegative_integer () {
+    local argument="$1"
+    if test "$argument" -ge 0 2>/dev/null ; then
+        # The arithmetic expansion removes a possible leading '+' in the output
+        # so that e.g. "is_nonnegative_integer +12" outputs '12' (and not '+12')
+        # and "is_nonnegative_integer -0" outputs '0' (and not '-0'):
+        echo $(( argument + 0 ))
```

Comment:
Why are we calling `echo` here?
When `is_nonnegative_integer` is not assigned to variable, final output might look like this:

```
2019-03-01 12:56:24.047800841 Verifying that the entries in /var/lib/rear/layout/disklayout.conf are correct ...
2019-03-01 12:56:24.048643873 Verifying that the 'disk' entries in /var/lib/rear/layout/disklayout.conf are correct
8589934592
2019-03-01 12:56:24.050638863 Verifying that the 'part' entries for /dev/sda in /var/lib/rear/layout/disklayout.conf are correct
209715200              <=== echoed by is_*_integer()  
1048576                  
0                            
210763776              
2936012800          
5579472896           
74448896               
8515485696            
73400320             
8516534272           
2019-03-01 12:56:24.052779940 Verifying that the 'part' entries for /dev/sda in /var/lib/rear/layout/disklayout.conf specify consecutive partitions
2019-03-01 12:56:24.056301632 /dev/sda2 size -1 is not a positive integer
2019-03-01 12:56:24.063888995 ERROR: 
====================
BUG in /usr/share/rear/layout/save/default/950_verify_disklayout_file.sh line 165:
'Entries in /var/lib/rear/layout/disklayout.conf are broken ('rear recover' would fail)'
--------------------
```

Same numbers are recorded in logfile, and one can wonder "what are these numbers?"

V.

Reply:
As I went through current ReaR code, `is_*_integer()` is never assigned to variable, so I'm assuming that `echo` is just some debugging leftover ?

V.

Reply:
Ah I've found this:

```
usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh:169:                check_interval=$( is_integer $check_interval )
```

So maybe thats where `echo` was born ...

V.